### PR TITLE
Drop requirement for OTC review

### PR DESCRIPTION
**Pull request by nhorman**
Current bylaws under general policies:
https://github.com/openssl/general-policies/blob/master/policies/committer-policy.md

Indicate the following:
```
All submissions must be reviewed and approved by at least two committers.
Neither of the reviewers can be the author of the submission.
```

Given that, it seems we should adjust our tooling to reflect the lack of need for an OTC approval

### CODE REVIEW

File: review-tools/gitaddrev
Comment (mattcaswell) [inline review-tools/gitaddrev:141]: Is this correct? I'm not necessarily against this change but previously only OTC could commit to this repo. Now any committer can. I'm not sure if that was the intent behind the bylaws change which I think was primarily about the main source repo - and I suppose other repos using the same rules as main source.

Comment (nhorman) [reply review-tools/gitaddrev:141]: I think its in keeping with the letter of they bylaws.  As to the spirit, I don't think I can say.  I'd look to other committers/OTC members  on that

Comment (t8m) [reply review-tools/gitaddrev:141]: This is the otc/tools repo in the primary GHE server. IMO this is wrong. The bylaws are about the primary openssl/openssl codebase but can be probably extended to all the openssl/<something> repos.

Please revert this hunk.

Comment (mattcaswell) [reply review-tools/gitaddrev:141]: > I think its in keeping with the letter of they bylaws

I don't think its in keeping with the letter either. It depends on what you think the scope of that sentence is about. But the policy in general talks about what committers can do, and committers are defined to be "contributors with commit access to the OpenSSL source code repository." - which I think is fairly unambiguously openssl/openssl. Other repos have different review and approval criteria which this policy does not cover.

**Comment by DDvO** [comment]
As mentioned to two project members already in emails on Sep 12 and 13:

With having only one 'review pending' label left on the GitHub GUI,
it is no more possible to quickly see from the labels (e.g., in the GitHub PR list view) whether one or two reviews are pending.
Moreover, 
now it can be rather unclear if a first valid approval is still there, because it may have been invalidated by further activity.
This can lead to confusion and misunderstandings on whether just one or even two approvals are still/again needed.

Therefore, **I propose having two such labels again**, e.g., 'review 1 pending' and 'review 2 pending'.

File: review-tools/gitaddrev
Comment (nhorman) [reply review-tools/gitaddrev:141]: so, whats the right change here?  I can agree that min_otc can/should be 2 for the tools repo, and 0 for openssl, what about perftools/installer/etc?

**Comment by tom-cosgrove-arm** [comment]
+1 to [@DDvO's comment](https://github.com/openssl/tools/pull/217#issuecomment-2383379778) - it's not clear to me now when is best to remove the approval pending label, having two labels was really useful

File: review-tools/gitaddrev
Comment (t8m) [reply review-tools/gitaddrev:141]: I think we can agree that perftools, installer and fuzz-corpora repos should have _similar_ rules as the main source code.

Comment (mattcaswell) [reply review-tools/gitaddrev:141]: Strictly speaking the committer policy that was changed *only* applies to openssl/openssl. However, I think the intent for installer was that it should have the same review requirements as openssl/openssl (although I don't know if that was documented anywhere....possibly in some meeting minutes somewhere). I don't know what the right answer for fuzz-corpora is - probably it needs separate discussion.

Comment (nhorman) [reply review-tools/gitaddrev:141]: So can we agree here that we should limit this change to the default case, as the other repos are explicitly called out?  Or do we need to bring this up at OTC?

Comment (mattcaswell) [reply review-tools/gitaddrev:141]: I think we limit it to just the main repo for now. All other repos require a separate agreement as to what the correct review criteria are.

**Review by mattcaswell — APPROVED** [review]

**Comment by DDvO** [comment]
> It is very easy to find out whether additional review is needed or not as GH indicates that via the merge blocked indicator. Or you can just count the green approval check marks.

I suppose you do not mean the "Merging is blocked" but "Review required" indicator, which automatically disappears when all two required reviews are present.
Yet this info cannot be seen from the list of open pull requests, 
unless the "review pending" label gets cleared and "review: done" get added - can this be achieved automatically? 

Moreover, counting the green approval check marks is tedious and can give wrong results due to superseded reviews.
And these are not shown in the list of open pull requests.
So quickly and correctly determining whether one or two reviews are currently needed is even less convenient/straightforward to do.

**Comment by t8m** [comment]
> Moreover, counting the green approval check marks is tedious and can give wrong results due to superseded reviews.

Oh really tedious? I consider it tedious to constantly change the labels for practically no gain.

Why do you need to distinguish PRs where two reviews are missing from PRs when only one review is missing?

When a review is superseded by something different than just  pushing new commits to the PR, it can be dismissed by any committer. However if the PR just needs changes the other option is to clear the review pending label. Of course that requires someone to re-add the label after the required change is pushed.

**Comment by t8m** [comment]
Anyway, this is off topic here.

**Review by t8m — APPROVED** [review]
Although I feel we could change also the installer, perftools and fuzz-corpora repos and keep the spirit of the bylaws change, I do not insist.

**Comment by t8m** [comment]
Merged. Thank you.

**Pull request closed without merging**
